docs/gateway-api-compatibility.md.md

```diff
@@ -0,0 +1,110 @@
+# Gateway API Compatibility
+
+This document describes which Gateway API resources NGINX Kubernetes Gateway supports and the extent of that support.
+
+## Summary
+
+| Resource | Support Status |
+|-|-|
+| [GatewayClass](#gatewayclass) | Partially supported |
+| [Gateway](#gateway) | Partially supported |
+| [HTTPRoute](#httproute) | Partially supported |
+| [TLSRoute](#tlsroute) | Not supported |
+| [TCPRoute](#tcproute) | Not supported |
+| [UDPRoute](#udproute) | Not supported |
+| [ReferenceGrant](#referencegrant) |  Not supported |
+| [Custom policies](#custom-policies) | Not supported |
+
+## Terminology
+
+The support status is described using the following words:
+- *Supported*. The resource or field is fully supported and conformant to the Gateway API specification.
+- *Partially supported*. The resource or field is supported partially or with limitations. It will become fully supported in the future releases.
+- *Not supported*. The resource or field is not yet supported. It will become partially or fully supported in the future releases.
+
+Note: it might be possible that NGINX Kubernetes Gateway will never support some resources and/or fields of the Gateway API. If that happens, that will be documented.
+
+## Resources
+
+Below we list the resources and the support status of their corresponding fields. 
+
+Note that we don't describe the meaning of each field. For that, please visit the [Gateway API documentation](https://gateway-api.sigs.k8s.io/references/spec/). 
+
+### GatewayClass 
+
+> Status: Partially supported. 
+
+NGINX Kubernetes Gateway supports only a single GatewayClass resource configured via `--gatewayclass` [cli argument](./cli-args.md).
+
+Fields:
+* `spec`
+	* `controllerName` - supported.
+	* `parametersRef` - not supported.
+	* `description` - supported.
+* `status` - partially supported.
+
+### Gateway
+
+> Status: Partially supported.
+
+NGINX Kubernetes Gateway supports only a single Gateway resource: it select the Gateway resource among all resources that belong to
+NGINX Kubernetes Gateway (i.e. reference NGINX Kubernetes Gateway corresponding GatewayClass). In case of multiple Gateway resources, NGINX Kubernetes Gateway will use a
+deterministic conflict resolution strategy: it will choose the oldest resource based on the creation timestamp. If the timestamps are equal, NGINX Kubernetes Gateway will chose the resource that appears alphabetically first based on its namespace and then name. We might support multiple Gateway resources. If you're interested in that support, please share your use case with us.
```

Comment:
```suggestion
deterministic conflict resolution strategy: it will choose the oldest resource by creation timestamp. If the timestamps are equal, NGINX Kubernetes Gateway will choose the resource that appears first in alphabetical order by “{namespace}/{name}”. We might support multiple Gateway resources. Please share your use case with us if you're interested in that support.
```